src/common/modes.c

```diff
@@ -765,7 +765,11 @@ handle_mode (server * serv, char *word[], char *word_eol[],
 			if ((all_modes_have_args || mode_has_arg (serv, sign, *modes)) && arg < (num_args + 1))
 			{
 				arg++;
-				argstr = word[arg + offset];
+				// is this the last argument?
```

Comment:
Could you elaborate on this. This seems to be a behavior change beyond the trailing `:`

Reply:
This is to remove the trailing colon in a message like `MODE #channel +v user user :user`

Reply:
That seems wrong.

The first arg that starts with `:` becomes the trailing arg.

Reply:
I think combining it with a check in the arg_count loop for `:` should work, that would ensure the last argument is always the trailing argument and would handle it all as one

Reply:
Sure I think it will work but it is a violation of the protocol and since this whole discussion is about that I don't think it should be done.

Reply:
in what way? the check in the loop would make it so the argument with the `:` would be considered the last arg and it would use word_eol for that. Having a trailing parameter on `MODE` is valid

Reply:
InspIRCd v3 sends `MODE #channel +oo user :user1`, each mode parameter being a new command parameter, and the last one being a trailing param. The check in the loop ensures the arg_count reflects the actual number (ie 2 for `MODE +gg word :multi word`)

Reply:
> The check in the loop ensures the arg_count reflects the actual number (ie 2 for MODE +gg word :multi word)

Yes but that ensuring is incorrect. `MODE +gggg :a b c d` According the protocol that is `a b c d` not 4 args.

Reply:
hmm, I see you fixed `num_args` above. So assuming that is all correct why does the `if (arg == num_args + 1)` check need to happen? It would be safe to call `STRIP_COLON` each arg.